src/peft/tuners/lora.py

```diff
@@ -580,15 +579,22 @@ def _unload_and_optionally_merge(self, merge=True, progressbar: bool = False):
 
         return self.model
 
-    def add_weighted_adapter(self, adapters, weights, adapter_name, combination_type="svd"):
+    def add_weighted_adapter(
+        self, adapters, weights, adapter_name, combination_type="svd", svd_rank=None, svd_params={}, svd_clamp=None
```

Comment:
Let's avoid having a mutable default argument, `svd_params={}`. Same in `_svd_weighted_adapter`. How about instead, we just collect the `**kwargs` here and pass them along. That way, we can also potentially use them in the future if there are other combination types with `kwargs`.

Reply:
Actually, there are just two meaningful parameters, which from my perspective should be available for SVD - `driver` and `full_matrices` - the same as in [`torch.linalg.svd`](https://pytorch.org/docs/stable/generated/torch.linalg.svd.html) (they both may influence SVD decomposition speed). Additionally, here we introduce `svd_rank` which controls output LoRA rank, and `svd_clamp`, which controls clamping quantile or clamping disabling.

Also, using kwargs in case of SVD can be error-prone if somebody tries to pass the parameter named `out` - there will be an error, because [`torch.linalg.svd`](https://pytorch.org/docs/stable/generated/torch.linalg.svd.html) uses `out` to store matrices after decomposition.

Maybe it's better to make all these parameters explicit and prefix them with `combination_type`? So, the documentation will become clear for those who attempts to use this method.
So, for SVD we will have the following parameters:

- `svd_rank` - our introduced parameter - output rank after LoRA decomposition
- `svd_clamp` - out introduced parameter - whether to perform clamping of SVD matrices, or clamping quantile
- `svd_driver` - `torch.linalg.svd` driver parameter - cuSOLVER algorithm to use.
- `svd_full_matrices` - `torch.linalg.svd` `full_matrices` parameter - whether to perform full or reduced SVD decomposition

And, in case we need to introduce some new parameters we can prefix them with `combination_type` (e.g. `cat_param1`, `cat_param2`, `linear_param1`, ...). The only downside I see, is that in case `torch` changes the signature for `torch.linalg.svd` (which seems very unrealistic), the code here needs to be updated to match latest parameters.

What do you think?

Reply:
Ok, refactored it in one of the latest commits.